tests/kafkatest/tests/consumer_test.py

```diff
@@ -0,0 +1,143 @@
+from kafkatest.services.kafka import KafkaService
+from kafkatest.services.verifiable_producer import VerifiableProducer
+from kafkatest.services.verifiable_consumer import VerifiableConsumer, TopicPartition
+
+import time
+
+def partitions_for(topic, num_partitions):
+    partitions = set()
+    for i in range(num_partitions):
+        partitions.add(TopicPartition(topic=topic, partition=i))
+    return partitions
+
+class VerifiableConsumerTest(KafkaTest):
+
+    STOPIC = "simple_topic"
+    TOPIC = "test_topic"
+    NUM_PARTITIONS = 3
+    PARTITIONS = partitions_for(TOPIC, NUM_PARTITIONS)
+    GROUP_ID = "test_group_id"
+
+    def __init__(self, test_context):
+        super(VerifiableConsumerTest, self).__init__(test_context, num_zk=1, num_brokers=2, topics={
+            self.TOPIC : { 'partitions': self.NUM_PARTITIONS, 'replication-factor': 1 },
+            self.STOPIC : { 'partitions': 1, 'replication-factor': 2 }
+        })
+        self.num_producers = 1
+        self.num_consumers = 2
+
+    def min_cluster_size(self):
+        """Override this since we're adding services outside of the constructor"""
+        return super(VerifiableConsumerTest, self).min_cluster_size() + self.num_consumers + self.num_producers
+
+    def _partitions(self, assignment):
+        partitions = []
+        for parts in assignment.itervalues():
+            partitions += parts
+        return partitions
+
+    def _valid_assignment(self, assignment):
+        partitions = self._partitions(assignment)
+        return len(partitions) == self.NUM_PARTITIONS and set(partitions) == self.PARTITIONS
+
+    def _setup_consumer(self, topic):
+        return VerifiableConsumer(self.test_context, self.num_consumers, self.kafka,
+                                  topic, self.GROUP_ID, session_timeout=10000)
+
+    def _setup_producer(self, topic, max_messages=-1):
+        return VerifiableProducer(self.test_context, self.num_producers,
+                                  self.kafka, topic, max_messages=max_messages)
+
+    def _await_all_members(self, consumer):
+        # Wait until all members have joined the group
+        wait_until(lambda: len(consumer.joined) == self.num_consumers, timeout_sec=20,
+                   err_msg="Consumers failed to join in a reasonable amount of time")
+
+    def test_consumer_failure(self):
+        partition = TopicPartition(self.STOPIC, 0)
+        
+        consumer = self._setup_consumer(self.STOPIC)
+        producer = self._setup_producer(self.STOPIC)
+
+        consumer.start()
+        self._await_all_members(consumer)
+
+        partition_owner = consumer.owner(partition)
+        assert partition_owner is not None
+
+        producer.start()
+        wait_until(lambda: producer.num_acked > 1000, timeout_sec=20,
+                   err_msg="Producer failed waiting for messages to be written")
+
+        consumer.kill_consumer(partition_owner)
+        time.sleep(10)
```

Comment:
Ideally it's best if we can avoid `sleep` calls
Is there a reason why you can't use `stop_node` without sleeping? This will block until the process is gone and the background thread has finished processing output from the consumer?